x-pack/filebeat/input/s3/input.go

```diff
@@ -436,7 +437,25 @@ func (p *s3Input) createEventsFromS3Info(svc s3iface.ClientAPI, info s3Info, s3C
 
 	reader := bufio.NewReader(resp.Body)
 
-	// Check if expand_event_list_from_field is given with document conent-type = "application/json"
+	isS3ObjGzipped, err := isStreamGzipped(reader)
+	if err != nil {
+		err = errors.Wrap(err, "could not determine if S3 object is gzipped")
+		p.logger.Error(err)
+		return err
+	}
+
+	if isS3ObjGzipped {
+		gzipReader, err := gzip.NewReader(reader)
+		if err != nil {
+			err = errors.Wrap(err, "gzip.NewReader failed")
+			p.logger.Error(err)
+			return err
+		}
+		reader = bufio.NewReader(gzipReader)
+		gzipReader.Close()
```

Comment:
Interesting that `reader` as `bufio.NewReader(gzipReader)` continues working after closing the `gzipReader`.

Reply:
Yes, I found this interesting as well. 🤔